homeassistant/components/config/scene.py

```diff
@@ -0,0 +1,65 @@
+"""Provide configuration end points for Scenes."""
+from collections import OrderedDict
+import uuid
+
+from homeassistant.components.scene import DOMAIN, PLATFORM_SCHEMA
+from homeassistant.const import CONF_ID, SERVICE_RELOAD
+from homeassistant.config import SCENE_CONFIG_PATH
+import homeassistant.helpers.config_validation as cv
+
+from . import EditIdBasedConfigView
+
+
+async def async_setup(hass):
+    """Set up the Scene config API."""
+
+    async def hook(hass):
+        """post_write_hook for Config View that reloads scenes."""
+        await hass.services.async_call(DOMAIN, SERVICE_RELOAD)
+
+    hass.http.register_view(
+        EditSceneConfigView(
+            DOMAIN,
+            "config",
+            SCENE_CONFIG_PATH,
+            cv.string,
+            PLATFORM_SCHEMA,
+            post_write_hook=hook,
+        )
+    )
+    return True
+
+
+class EditSceneConfigView(EditIdBasedConfigView):
+    """Edit scene config."""
+
+    def _write_value(self, hass, data, config_key, new_value):
+        """Set value."""
+        index = None
+        for index, cur_value in enumerate(data):
+            # When people copy paste their scenes to the config file,
+            # they sometimes forget to add IDs. Fix it here.
+            if CONF_ID not in cur_value:
+                cur_value[CONF_ID] = uuid.uuid4().hex
+
+            elif cur_value[CONF_ID] == config_key:
+                break
+        else:
+            cur_value = OrderedDict()
```

Comment:
Dicts are ordered by default in Python 3.6+.

Reply:
I'm not getting the same result using a `dict`

Reply:
Write a test and show us :)

Reply:
![image](https://user-images.githubusercontent.com/5662298/68089326-fb236c00-fe67-11e9-844f-a645ea8fe2db.png)

The order of the dict seems ok, but I think `PyYAML` can't deal with it? The YAML is not sorted correctly.